chrome/content/zotero/xpcom/data/item.js

```diff
@@ -3955,6 +3955,28 @@ Zotero.Item.prototype.getAnnotations = function (includeTrashed) {
 };
 
 
+/**
+ * Determine if the item is a PDF attachment that exists on disk and contains
+ * embedded annotations.
+ *
+ * @return {Promise<Boolean>}
+ */
+Zotero.Item.prototype.hasEmbeddedAnnotations = async function () {
+	if (!this.isPDFAttachment()) {
+		return false;
+	}
+
+	let path = await this.getFilePathAsync();
+	if (!path) {
+		return false;
+	}
+
+	let contents = await Zotero.File.getContentsAsync(path);
+	let re = /\/Type\s+\/Annot\s/;
```

Comment:
I'm concerned about link annotations that are quite common in PDFs. Maybe we should test for specific annotation types instead. Or subtract link annotation count from the total annotations count.

Reply:
OK, see latest commit. A slightly more complicated parsing routine but should still be pretty fast.

Reply:
Wait, but why not just test if `Highlight`, `Underline` and other markup annotation subtypes just exist in the file, with one regex line?

Reply:
Oh, that's a great question. Don't know why I didn't think of that. Can those ever be subtypes of another object type? I guess probably not.

Reply:
OK, fixed. Thanks so much for pointing that out.

Reply:
Looks good!